sdk/cognitiveservices/azure-cognitiveservices-inkrecognizer/azure/cognitiveservices/inkrecognizer/_client.py

```diff
@@ -0,0 +1,319 @@
+        if self.response_hook is not None:
+            if not callable(self.response_hook):
+                raise ValueError("response_hook should be callable.")
+        timeout = _get_and_validate_param(kwargs, "timeout", factory=float)
+        client_request_id = _get_and_validate_param(kwargs, "client_request_id", factory=str)
+        self.headers = _get_and_validate_param(kwargs, "headers", default={})
+        self.headers["x-ms-client-request-id"] = client_request_id or str(uuid.uuid4())  # generate random uuid
+        if timeout is not None:
+            kwargs["connection_timeout"] = timeout
+        self.kwargs = kwargs
+
+
+class _InkRecognizerClientBase(object):
+    def __init__(self, url, credential, **kwargs):
+        # type: (str, TokenCredential, Any) -> None
+
+        self._url = str(url)
+        self._default_arguments = _DEFAULT_ARGUMENTS.copy()
+        self._default_arguments.update(kwargs)
+
+        azure_config = _AzureConfiguration(credential, **kwargs)
+        self._pipeline_client = PipelineClient(
+            base_url=url, config=azure_config, transport=azure_config.transport)
+
+    def __enter__(self):
+        self._pipeline_client.__enter__()
+        return self
+
+    def __exit__(self, *exc_details):
+        self._pipeline_client.__exit__(*exc_details)
+
+    def _generate_config(self, request_arguments):
+        arguments = self._default_arguments.copy()
+        arguments.update(request_arguments)
+        config = _InkRecognizerConfiguration(**arguments)
+        return config
+
+    def _generate_url(self, config):
+        return self._url + config.service_version._to_string()  # pylint:disable=protected-access
+
+    def _pack_one_stroke(self, stroke):  # pylint:disable=no-self-use
+        stroke_points = [{"x": point.x, "y": point.y} for point in stroke.points]
+        stroke_json = {
+            "id": stroke.id,
+            "points": stroke_points,
+        }
+        if stroke.kind in [InkStrokeKind.DRAWING, InkStrokeKind.WRITING]:
+            stroke_json["kind"] = stroke.kind.value
+        if stroke.language:
+            stroke_json["language"] = stroke.language
+        return stroke_json
+
+    def _pack_request(self, ink_stroke_list, config):
+        return json.dumps({
+            "applicationType": config.application_kind.value,
+            "language": config.language,
+            "unit": config.ink_point_unit.value,
+            "unitMultiple": config.unit_multiple,
+            "strokes": [self._pack_one_stroke(stroke)
+                        for stroke in ink_stroke_list
+                        if len(stroke.points) != 0]
+        })
+
+    def _parse_result(self, response, config):  # pylint:disable=inconsistent-return-statements
+        status_code = response.status_code
+        headers = response.headers
+        content = response.body().decode("utf-8")
+
+        if status_code == 200:
+            logging.info("Response headers:")
```

Comment:
Don't log, the pipeline will do it depending on a global configuration

Reply:
Removed logging for request headers, request body, response headers, response body.